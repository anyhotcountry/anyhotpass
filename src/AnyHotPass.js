import React, { useState, useEffect, useRef } from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import generatePass from 'anyhotpass-lib';

function AnyHotPass() {
  const [master, setMaster] = useState('');
  const [domain, setDomain] = useState('');
  const [password, setPassword] = useState('');
  const masterRef = useRef(null);
  const passwordRef = useRef(null);

  useEffect(() => {
    masterRef.current.focus();      
  }, []);

  const generate = (e) => {
    e.preventDefault();
    if (master.length === 0 || domain.length === 0) {
      return;
    }
    const newPassword = generatePass(master, domain, 18);
    setPassword(newPassword);
    passwordRef.current.focus();
  };

  return (
    <>
      <Modal show="true" backdrop="false">
        <Modal.Header>
          <Modal.Title>AHP</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form>
            <Form.Group controlId="formPassword">
              <Form.Control type="password" ref={masterRef} placeholder="Master Password" required onChange={(e) => setMaster(e.target.value)} />
            </Form.Group>
            <Form.Group controlId="formDomain">
              <Form.Control type="url" placeholder="example.com" required onChange={(e) => setDomain(e.target.value)} />
            </Form.Group>
            <Button variant="secondary" type="submit" block onClick={(e) => generate(e)} class="mr-1" >Generate</Button>
            <Form.Group controlId="formDomain" className="mt-2">
              <Form.Control type="text" ref={passwordRef} readOnly onFocus={(e) => e.target.select()} value={password} />
            </Form.Group>
          </Form>
        </Modal.Body>
      </Modal>
    </>
  )
}

export default AnyHotPass;
