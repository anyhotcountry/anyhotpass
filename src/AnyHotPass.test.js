import React from 'react';
import { render } from '@testing-library/react';
import AnyHotPass from './AnyHotPass';

test('AnyHotPass render matches snapshot', () => {
  const { asFragment } = render(<AnyHotPass />);
  expect(asFragment()).toMatchSnapshot();
});
