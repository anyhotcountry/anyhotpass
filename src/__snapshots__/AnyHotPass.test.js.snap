// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnyHotPass render matches snapshot 1`] = `<DocumentFragment />`;
